src/core/mod.rs

```diff
@@ -307,6 +307,24 @@ impl BPF {
         }
     }
 
+    pub fn get_syscall_prefix(&mut self) -> Result<String, Error> {
```

Comment:
You're returning a `Result` but the implementation seems to never return an error? I believe the implementation is based on https://github.com/iovisor/bcc/blob/ab54de2e4449027f2b4dccd022adc57bec4fd4eb/src/python/bcc/__init__.py#L623

I'm not sure I entirely agree with returning just the first prefix from the slice. I wonder if there are cases where this would not be the right thing to return, in which case I'd rather return an `Err` instead.

Eitherway - either the return type should always be String and always result in something that works, or we should leave it to the user to handle the `Result` type themselves. I wonder if an enum with a default variant would make a better API surface here.

@jvns - do you have thoughts on this? We can make it like the python and just return a best effort string or decide on a better API.

Reply:
Resorted to just returning a `String`, happy to rework based on this conversation.